### PR TITLE
192 update the chart data table

### LINKED ISSUE #192: Update the chart data table
TODO:

* Remove the percentage columns.
* For each value/label pair, if there are targets for the site display `[VALUE] / [TARGET] ([PERCENT]%)`. If there are no targets for the site then display `[VALUE] ([PERCENT]%)`.
* Add a "Totals" column that combines values and targets.

### PR DESCRIPTION
**Pull request by Elkrival**
Update Chart Data Table

Screen Shot 2022-10-04 at 5 03 17 PM

### CODE REVIEW

File: views/fe-utils/graphTableRowDataUtil.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { TOTAL_LABEL } from '../../constants'
+import { studyCountsToPercentage } from './helpers'
+
+export const graphTableRowData = (site) => {
+  const rowData = []
+  site.forEach((site) => {
```
Comment (mikestone14) [inline views/fe-utils/graphTableRowDataUtil.js]: there are 2 variables named `site` so we will want to change one of them.

File: views/fe-utils/graphTableRowDataUtil.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { TOTAL_LABEL } from '../../constants'
+import { studyCountsToPercentage } from './helpers'
+
+export const graphTableRowData = (site) => {
```
Comment (mikestone14) [inline views/fe-utils/graphTableRowDataUtil.js]: this is not a site but graph data by site, right?

Comment (mikestone14) [reply views/fe-utils/graphTableRowDataUtil.js]: this is a very specific helper function (it generates site-specific table row data for a bar chart) with a generic name in a generic folder. I think we want this to be more specifically named or more generally named but placed in a specific location, like in the `GraphTable` directory.

Comment (Elkrival) [reply views/fe-utils/graphTableRowDataUtil.js]: https://github.com/AMP-SCZ/dpdash/pull/193/commits/32c8504a19d74b69df56ae08b713b68c8eda7771

File: views/fe-utils/graphTableRowDataUtil.js (hunk 1 above)
Comment (Elkrival) [reply views/fe-utils/graphTableRowDataUtil.js]: https://github.com/AMP-SCZ/dpdash/pull/193/commits/32c8504a19d74b69df56ae08b713b68c8eda7771

File: views/ViewChart.react.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ import { Typography } from '@material-ui/core'
 import { chartStyles } from './styles/chart_styles'
 import AppLayout from './layouts/AppLayout'
 import BarGraph from './components/Graphs/BarGraph'
-import GraphTable from './components/Graphs/GraphTable'
+import GraphTable from './components/GraphTable/GraphTable'
```
Comment (mikestone14) [inline views/ViewChart.react.js]: the component should live in an `index.jsx` file so we don't need to repeat the name here.

File: views/components/GraphTable/GraphTable.jsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  const [totalsSite, ...sites] = graphTableRowDataBySiteUtil(graph.dataBySite)
+
+  return (
+    <Paper>
+      <Table>
+        <TableHead>
+          <TableRow>
+            <TableCell align="center">Site</TableCell>
+            {tableColumns.map(({ name }) => (
+              <React.Fragment key={name + 'table-header'}>
```
Comment (mikestone14) [inline views/components/GraphTable/GraphTable.jsx]: these keys don't need to be globally unique just unique to this map so we can use the names as the key.

```suggestion
              <React.Fragment key={name}>
```

File: views/components/GraphTable/GraphTable.jsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+                <TableCell align="center">{name}</TableCell>
+              </React.Fragment>
```
Comment (mikestone14) [inline views/components/GraphTable/GraphTable.jsx]: We don't need to use a fragment here.

```suggestion
              <TableCell align="center" key={name}>{name}</TableCell>
```

File: views/components/GraphTable/GraphTable.jsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+                    counts: { [columnLabel]: siteCount },
+                    targets: { [columnLabel]: siteTarget },
+                    percentages: { [columnLabel]: sitePercentage },
+                  } = site
+                  return (
+                    <React.Fragment key={columnLabel + 'site-row'}>
+                      <TableCell align="center" style={{ color }}>
+                        {formatSiteData(siteCount, siteTarget, sitePercentage)}
+                      </TableCell>
+                    </React.Fragment>
```
Comment (mikestone14) [inline views/components/GraphTable/GraphTable.jsx]: No need for a fragment here.

```suggestion
                    <TableCell align="center" key={columnLabel} style={{ color }}>
                      {formatSiteData(siteCount, siteTarget, sitePercentage)}
                    </TableCell>
```

File: views/components/GraphTable/GraphTable.jsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+                counts: { [columnLabel]: siteCount },
+                targets: { [columnLabel]: siteTarget },
+                percentages: { [columnLabel]: sitePercentage },
+              } = totalsSite
+              return (
+                <React.Fragment key={columnLabel + 'totals-row'}>
+                  <TableCell align="center" style={{ color }}>
+                    {formatSiteData(siteCount, siteTarget, sitePercentage)}
+                  </TableCell>
+                </React.Fragment>
```
Comment (mikestone14) [inline views/components/GraphTable/GraphTable.jsx]:
```suggestion
                <TableCell align="center" key={columnLabel} style={{ color }}>
                  {formatSiteData(siteCount, siteTarget, sitePercentage)}
                </TableCell>
```

File: views/components/GraphTable/helpers.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  const rowData = []
+  dataBySite.forEach((site) => {
+    const { count, targetTotal } = site.totalsForStudy
+    site.counts[TOTAL_LABEL] = count
+    site.targets[TOTAL_LABEL] = targetTotal
+    site.percentages[TOTAL_LABEL] = studyCountsToPercentage(count, targetTotal)
+    rowData.push(site)
+  })
+
+  return rowData
```
Comment (mikestone14) [inline views/components/GraphTable/helpers.js]:
```suggestion
  return dataBySite.map((site) => {
    const { count, targetTotal } = site.totalsForStudy
    site.counts[TOTAL_LABEL] = count
    site.targets[TOTAL_LABEL] = targetTotal
    site.percentages[TOTAL_LABEL] = studyCountsToPercentage(count, targetTotal)
    rowData.push(site)
  })
```

File: views/components/GraphTable/helpers.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import { TOTAL_LABEL } from '../../../constants'
+import { studyCountsToPercentage } from '../../fe-utils/helpers'
+
+export const graphTableRowDataBySiteUtil = (dataBySite) => {
```
Comment (mikestone14) [inline views/components/GraphTable/helpers.js]:
```suggestion
export const graphTableRowDataBySite = (dataBySite) => {
```

File: views/components/GraphTable/GraphTable.jsx [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+  Paper,
+} from '@material-ui/core'
+import { formatSiteData, graphTableColumns } from '../../fe-utils/helpers'
+import { graphTableRowDataBySiteUtil } from './helpers'
+
+const GraphTable = ({ graph }) => {
+  if (!graph) return null
+
+  const tableColumns = graphTableColumns(graph.labels)
+  const [totalsSite, ...sites] = graphTableRowDataBySiteUtil(graph.dataBySite)
```
Comment (mikestone14) [inline views/components/GraphTable/GraphTable.jsx]: how do we know the `totalsSite` is going to be first? Is that by coincidence? We may need logic in this helper function to ensure that's the case.

Comment (Elkrival) [reply views/components/GraphTable/GraphTable.jsx]: From your comment above I sorted the new site list and added logic to put Totals last, this way there's no deconstruction and just one map to render the data.
https://github.com/AMP-SCZ/dpdash/pull/193/commits/325e6635a9f09c0b3ce28c6d2a5ab8bfa1369601

File: views/components/GraphTable/helpers.js (hunk 9 above)
Comment (Elkrival) [reply views/components/GraphTable/helpers.js]: gotcha this doesn't need the Util appended
https://github.com/AMP-SCZ/dpdash/pull/193/commits/325e6635a9f09c0b3ce28c6d2a5ab8bfa1369601

File: views/components/GraphTable/helpers.js (hunk 8 above)
Comment (Elkrival) [reply views/components/GraphTable/helpers.js]: Thank you 
https://github.com/AMP-SCZ/dpdash/pull/193/commits/325e6635a9f09c0b3ce28c6d2a5ab8bfa1369601

File: views/components/GraphTable/GraphTable.jsx (hunk 7 above)
Comment (Elkrival) [reply views/components/GraphTable/GraphTable.jsx]: https://github.com/AMP-SCZ/dpdash/pull/193/commits/325e6635a9f09c0b3ce28c6d2a5ab8bfa1369601

File: views/ViewChart.react.js (hunk 3 above)
Comment (Elkrival) [reply views/ViewChart.react.js]: https://github.com/AMP-SCZ/dpdash/pull/193/commits/325e6635a9f09c0b3ce28c6d2a5ab8bfa1369601

File: views/components/GraphTable/GraphTable.jsx (hunk 6 above)
Comment (Elkrival) [reply views/components/GraphTable/GraphTable.jsx]: https://github.com/AMP-SCZ/dpdash/pull/193/commits/325e6635a9f09c0b3ce28c6d2a5ab8bfa1369601

File: views/components/GraphTable/GraphTable.jsx (hunk 4 above)
Comment (Elkrival) [reply views/components/GraphTable/GraphTable.jsx]: https://github.com/AMP-SCZ/dpdash/pull/193/commits/b5b1019b81de07f4778b4ec32f3e7bcecfc0ab17

File: views/components/GraphTable/GraphTable.jsx (hunk 5 above)
Comment (Elkrival) [reply views/components/GraphTable/GraphTable.jsx]: https://github.com/AMP-SCZ/dpdash/pull/193/commits/98ef1ad36740b7dcf6eae8a2e3f687620ac21cc2

File: views/ViewChart.react.js [hunk 11]
```diff
@@ -7,7 +7,7 @@ import { Typography } from '@material-ui/core'
 import { chartStyles } from './styles/chart_styles'
 import AppLayout from './layouts/AppLayout'
 import BarGraph from './components/Graphs/BarGraph'
-import GraphTable from './components/Graphs/GraphTable'
+import GraphTable from './components/GraphTable/'
```
Comment (mikestone14) [inline views/ViewChart.react.js]:
```suggestion
import GraphTable from './components/GraphTable'
```

**Review by mikestone14 — APPROVED** [review]
Looks good just one minor nit

**Pull request merged**